[JIT] CSE across context managers

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #76075

Fix for https://github.com/pytorch/pytorch/issues/75788

If one node is computing under `torch.no_grad` and the other is not, we shouldn't eliminate one of them.

Commit:
Dnt CSE across context managers

[ghstack-poisoned]

Commit:
Update on "[JIT] CSE across context managers"


Fix for https://github.com/pytorch/pytorch/issues/75788

If one node is computing under `torch.no_grad` and the other is not, we shouldn't eliminate one of them. 

[ghstack-poisoned]

Commit:
Update on "[JIT] CSE across context managers"


Fix for https://github.com/pytorch/pytorch/issues/75788

If one node is computing under `torch.no_grad` and the other is not, we shouldn't eliminate one of them. 

[ghstack-poisoned]

davidberard98 reviewed (approved):
LGTM! probably good to rebase over the build failures just to check

Commit:
Update on "[JIT] CSE across context managers"


Fix for https://github.com/pytorch/pytorch/issues/75788

If one node is computing under `torch.no_grad` and the other is not, we shouldn't eliminate one of them. 

[ghstack-poisoned]

Commit:
Update on "[JIT] CSE across context managers"


Fix for https://github.com/pytorch/pytorch/issues/75788

If one node is computing under `torch.no_grad` and the other is not, we shouldn't eliminate one of them. 

[ghstack-poisoned]

eellison commented:
@pytorchbot merge this please

seemethere commented:
@pytorchbot revert this, broke [internal xplat build](https://www.internalfb.com/intern/sandcastle/job/9007199894209948/artifact/actionlog.9007205940281931.log?selectedLines=44057-44057-24-68) and is blocking diff train from landing
